packages/manager/apps/pci-databases-analytics/src/pages/services/create/_components/ApiTerraformDialog.component.tsx

```diff
@@ -0,0 +1,315 @@
+    setTerraformData,
+  ] = useState<TTerraform.cloud.Response | null>(null);
+  const { serviceToTerraform } = useServiceToTerraform({
+    onError: () => {
+      toast({
+        title: t('pciDatabasesAddCommandDialogTerraformFetchError'),
+        variant: 'destructive',
+      });
+    },
+    onSuccess: (data) => {
+      setTerraformData(data);
+    },
+  });
+
+  type TabType = 'api' | 'terraform';
+  const [currentTab, setCurrentTab] = useState<TabType>('api');
+  const handleTabChange = async (tab: TabType) => {
+    setCurrentTab(tab);
+    if (tab === 'terraform' && !terraformData) {
+      // Handle Terraform data
+      const nodesList: TTerraform.cloud.database.service.Node[] = [];
+      // Add nodes to nodesList depending on data.nbNodes
+      for (let i = 0; i < dialogData.nodes; i += 1) {
+        nodesList.push({
+          flavor: dialogData.availability.flavor,
+          region: dialogData.availability.region,
+        });
+      }
+
+      const terraformService: TTerraform.cloud.database.ServiceRequest = {
+        body: {
+          description: dialogData.name,
+          nodesList,
+          plan: dialogData.availability.plan,
+          version: `${dialogData.availability.version}`,
+          ipRestrictions: dialogData.ipRestrictions,
+        },
+        pathParams: {
+          engine: dialogData.availability
+            .engine as TTerraform.cloud.database.EngineEnum,
+          serviceName: projectId,
+        },
+      };
+      if (dialogData.flavor.storage) {
+        terraformService.body.disk = {
+          size:
+            dialogData.flavor.storage.minimum.value +
+            dialogData.additionalStorage,
+        } as Disk;
+      }
+      serviceToTerraform(terraformService);
+    }
+  };
+
+  // Fill API configuration
+  const apiBody: ServiceCreationWithEngine = {
+    description: dialogData.name,
+    engine: dialogData.availability.engine as database.EngineEnum,
+    nodesPattern: {
+      flavor: dialogData.availability.flavor,
+      number: dialogData.nodes,
+      region: dialogData.availability.region,
+    },
+    plan: dialogData.availability.plan,
+    version: dialogData.availability.version,
+    ipRestrictions: dialogData.ipRestrictions,
+  };
+  if (dialogData.network.type === database.NetworkTypeEnum.private) {
+    const networkOpenstackId = dialogData.network.network.regions.find((r) =>
+      r.region.includes(dialogData.availability.region),
+    ).openstackId;
+    apiBody.networkId = networkOpenstackId;
+    apiBody.subnetId = dialogData.network.subnet.id;
+  }
+  if (dialogData.flavor.storage) {
+    apiBody.disk = {
+      size:
+        dialogData.flavor.storage.minimum.value + dialogData.additionalStorage,
+    } as database.service.Disk;
+  }
+  // Destructure body to extract engine and keep the rest
+  const { engine, ...formattedApiBody } = apiBody;
+  const apiEndpoint = `/cloud/project/${projectId}/database/${engine}`;
+  const apiBodyContent = formattedApiBody;
```

Comment:
It seems you are adding the engine in the apibody to later remove it
```suggestion
  // Fill API configuration
  const apiBodyContent = {
    description: dialogData.name,
    nodesPattern: {
      flavor: dialogData.availability.flavor,
      number: dialogData.nodes,
      region: dialogData.availability.region,
    },
    plan: dialogData.availability.plan,
    version: dialogData.availability.version,
    ipRestrictions: dialogData.ipRestrictions,
  } as database.ServiceCreation;
  if (dialogData.network.type === database.NetworkTypeEnum.private) {
    const networkOpenstackId = dialogData.network.network.regions.find((r) =>
      r.region.includes(dialogData.availability.region),
    ).openstackId;
    apiBodyContent.networkId = networkOpenstackId;
    apiBodyContent.subnetId = dialogData.network.subnet.id;
  }
  if (dialogData.flavor.storage) {
    apiBodyContent.disk = {
      size:
        dialogData.flavor.storage.minimum.value + dialogData.additionalStorage,
    } as database.service.Disk;
  }
  const apiEndpoint = `/cloud/project/${projectId}/database/${dialogData.availability.engine}`;
```